core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -106,7 +112,7 @@ public String toString() {
   protected class BaseMetastoreCatalogTableBuilder implements TableBuilder {
     private final TableIdentifier identifier;
     private final Schema schema;
-    private final ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+    private final Map<String, String> tableProperties = Maps.newHashMap();
```

Comment:
Is this needed for null-values? If so, can we ensure that we have tests that include null values? Otherwise it's likely to get refactored at some point.

Reply:
I don't think we will need null values.

Reply:
It's mainly because the map gets altered with `tableProperties.putAll(tableDefaultProperties());` and `tableProperties.putAll(tableOverrideProperties());`